src/backend/services/auth/basic.py

```diff
@@ -0,0 +1,89 @@
+from typing import List
+
+import bcrypt
+from sqlalchemy.orm import Session
+
+from backend.models.user import User
+from backend.services.auth.base import BaseAuthenticationStrategy
+
+
+class BasicAuthentication(BaseAuthenticationStrategy):
+    """
+    Basic email/password auth strategy.
+
+    Attributes:
+        NAME (str): The name of the strategy.
+        SHOULD_ATTACH_TO_APP (str): Whether the strategy needs to be attached to the FastAPI application.
+        SHOULD_AUTH_REDIRECT (str): Whether the strategy requires a redirect to the /auth endpoint after login.
+    """
+
+    NAME = "Basic"
+    SHOULD_ATTACH_TO_APP = False
+    SHOULD_AUTH_REDIRECT = False
+
+    @staticmethod
+    def get_required_payload() -> List[str]:
+        """
+        Retrieves the required /login payload for the Auth strategy.
+
+        Returns:
+            List[str]: List of required variables.
+        """
+        return ["email", "password"]
+
+    @staticmethod
+    def hash_and_salt_password(plain_text_password: str) -> str:
+        """
+        Hashes a given plain-text password with a randomly generated salt.
+
+        Args:
+            plain_text_password (str): Password to hash.
+
+        Returns:
+            str: Hashed password
+        """
+        return bcrypt.hashpw(plain_text_password.encode("utf-8"), bcrypt.gensalt())
+
+    @staticmethod
+    def check_password(plain_text_password: str, hashed_password: str) -> bool:
+        """
+        Hashes a given plain-text password with a randomly generated salt.
```

Comment:
i think this description might be an accidental copy paste from above

Reply:
ty